python/ray/data/datasource/lancedb_datasource.py

```diff
@@ -0,0 +1,87 @@
+"""
+Module to read an LanceDB table into a Ray Dataset, by using the Ray Datasource API.
+"""
+
+import logging
+from typing import List, Optional, Union
+
+import lance
+import pyarrow as pa
+from lance import LanceFragment
+
+from ray.data import ReadTask
+from ray.data.block import Block, BlockMetadata
+from ray.data.datasource import Datasource
+from ray.util.annotations import DeveloperAPI
+
+logger = logging.getLogger(__name__)
+
+
+@DeveloperAPI
+class LanceDBDatasource(Datasource):
+    """LanceDB Datasource
+    Read a LanceDB table as a Ray Dataset
+
+    Parameters
+    ----------
+    uri : str
+        The base URI of the LanceDB dataset.
+    columns: list
+        A list of columns to return from the dataset.
+    filter: str
+        A standard SQL expressions as predicates for dataset filtering.
+    """
+
+    def __init__(
+        self,
+        uri: str,
```

Comment:
It might be worth just supporting passing in a already configured `LanceDataset`. Then you don't have to reproduce all the same options.

Reply:
Yeah, either that or allowing `kwargs` to get passed down.